### DISCUSSION TITLE
overloading __getattr__()

### DISCUSSION THREAD

**Discussion by lovetox** [microsoft/pyright#2737] [Q&A] [answered]
Hi, is it possible to overload __getattr__()

```python3
from typing import Any
from typing import overload
from typing import Literal

class Obj:
    @overload
    def __getattr__(self, name: Literal["foo"]) -> int: ...
    @overload
    def __getattr__(self, name: Literal["bar"]) -> str: ...
    def __getattr__(self, name: str) -> Any:
        if name == 'foo':
            return 1
        return '1'

a = Obj()
b = a.foo
c = a.bar

b = getattr(a, 'foo')
c = getattr(a, 'bar')
```

so obviously foo and bar is not recognized by pyright as Literal, but i have no idea what i could use otherwise.

**Comment by erictraut** [comment, marked as answer]
When pyright evaluates `__getattr__`, it assumes that the `name` is of type `str`. It doesn't create a literal type for the member name. I think it's reasonable to support this. It's a relatively minor change. 

I'll point out that this doesn't work in mypy, which complains that the signature for the `__getattr__` overload is incorrect.

**Comment by lovetox** [comment]
Seems there is an issue on the mypy tracker, for reference: https://github.com/python/mypy/issues/11142

I found another workaround for my use case, so i don’t need this right now.
